nixos/modules/tasks/filesystems.nix

```diff
@@ -328,7 +331,9 @@ in
         )}
       '';
 
-    boot.initrd.systemd.contents."/etc/fstab".source = initrdFstab;
+    boot.initrd.systemd.storePaths = [initrdFstab];
+    boot.initrd.systemd.managerEnvironment.SYSTEMD_SYSROOT_FSTAB = initrdFstab;
+    boot.initrd.systemd.services.initrd-parse-etc.environment.SYSTEMD_SYSROOT_FSTAB = initrdFstab;
```

Comment:
@ElvishJerricco Could you share the reason for removing `/etc/fstab` from the initrd filesystem? I found that this broke one of my custom initrd units, which had a `mount <mountpoint>` line. I have a workaround, but I'm curious to understand the reason for this change so that I can choose the best workaround in my config.